#!/usr/bin/env python
PACKAGE = "geometric_controller"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# geometric_controller

gen.add("max_acc",    double_t,    0, "Max Feedback Acceleration", 0.1,  0.1, 8.0)
gen.add("Kp_x", double_t, 0, "Proportional gain for X position error", 1.0,  0.10, 10.0)
gen.add("Kp_y", double_t, 0, "Proportional gain for Y position error", 1.0, 0.1, 10.0)
gen.add("Kp_z", double_t, 0, "Proportional gain for Z position error", 2.0, 0.1, 10.0)
gen.add("Kv_x", double_t, 0, "Proportional gain for X velocity error", 1.5,  0.0, 10.0)
gen.add("Kv_y", double_t, 0, "Proportional gain for Y velocity error", 1.5, 0.0, 10.0)
gen.add("Kv_z", double_t, 0, "Proportional gain for Z velocity error", 2.5, 0.0, 10.0)
gen.add("Ki_x", double_t, 0, "Integral gain for X position error", 0.1,  0.0, 10.0)
gen.add("Ki_y", double_t, 0, "Integral gain for Y position error", 0.1, 0.0, 10.0)
gen.add("Ki_z", double_t, 0, "Integral gain for Z position error", 0.1, 0.0, 10.0)
gen.add("Krp", double_t, 0, "K roll pitch", 3.5, 0.1, 30.0)
gen.add("Kyaw", double_t, 0, "K yaw", 3.0, 0.1, 10.0)
gen.add("Thrust", double_t, 0, "Thrust_constant", 0.0450, 0.01, 0.1)
gen.add("K_xi",    double_t,    0, "xi_coefff", 1.5,  -3.0, 3.0)
gen.add("K_w",    double_t,    0, "omega_coeff", 1.5, -3.0, 3.0)
gen.add("Kpl_xy",    double_t,    0, "Proportional gain for load X position error", 2.0,  0.10, 10.0)
gen.add("Kpl_z", double_t, 0, "Proportional gain for loadZ position error", 3.0, 0.1, 10.0)
gen.add("Kvl_xy",    double_t,    0, "Proportional gain for load X velocity error", 6.0,  0.10, 10.0)
gen.add("Kvl_z", double_t, 0, "Proportional gain for Z velocity error", 6.0, 0.10, 10.0)
exit(gen.generate(PACKAGE, "geometric_controller", "GeometricController"))
